session.go

```diff
@@ -301,6 +302,14 @@ func (s *Session) Call(ctx context.Context, op interface{}, resp interface{}) er
 		return err
 	}
 
+	if len(reply.Errors) > 0 {
+		err := errors.New("rpc call returned errors")
+		for _, v := range reply.Errors {
+			err = errors.Join(err, fmt.Errorf("type: %s, tag: %s, appTag: %s, path: %s, message: %s", v.Type, v.Tag, v.AppTag, v.Path, v.Message))
```

Comment:
errors.Join only came out in 1.20 and we need to at least support 1.19 and maybe 1.18.  

But i do think that we should use it, but maybe backport it for older versions and guard it with build tags (it's pretty simple)

Also this should not be a fmt.Errorf which is just a string but use this type: https://github.com/nemith/netconf/blob/main/msg.go#L59

The Error method on this type makes it a Go error type that can be used.